addons/l10n_ro_edi/models/ciusro_document.py

```diff
@@ -0,0 +1,198 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import io
+import requests
+import zipfile
+
+from lxml import etree
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+NS_UPLOAD = {"ns": "mfp:anaf:dgti:spv:respUploadFisier:v1"}
+NS_STATUS = {"ns": "mfp:anaf:dgti:efactura:stareMesajFactura:v1"}
+NS_HEADER = {"ns": "mfp:anaf:dgti:efactura:mesajEroriFactuta:v1"}
+NS_SIGNATURE = {"ns": "http://www.w3.org/2000/09/xmldsig#"}
+
+
+class CIUSRODocument(models.Model):
+    _name = 'ciusro.document'
+    _description = "Document object for tracking CIUS-RO XML sent to E-Factura"
+    _order = 'datetime DESC, id DESC'
+
+    move_id = fields.Many2one('account.move', required=True)
+    state = fields.Selection(
+        selection=[
+            ('to_send', 'To send'),
+            ('sending', 'Sending'),
+            ('sent', 'Sent'),
+            ('nok', 'Not OK'),
+            ('ok', 'OK'),
+            ('error', 'Error'),
+        ],
+        string='E-Factura Status',
+        default='to_send',
+    )
+    attachment_id = fields.Binary(string='E-Factura Attachment')
+    datetime = fields.Datetime()
+    message = fields.Char()
+    key_loading = fields.Char()
+    key_download = fields.Char()
+    key_signature = fields.Char()
+    key_certificate = fields.Char()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        documents = super().create(vals_list)
+        for document in documents:
+            document.move_id.l10n_ro_edi_document_ids |= document
+            document.move_id.l10n_ro_edi_active_document_id = document
+        return documents
+
+    def _get_attachment_file_name(self):
+        self.ensure_one()
+        return f"{self.move_id.name.replace('/', '_')}_{self.id}.zip"
+
+    def _handle_error(self, message: str):
+        self.state = 'error'
+        self.message = message
```

Comment:
I don't like the design of this document. The document should be created once during the process with all values filled and shouldn't evolve over the time imo. It's a one shot:
Make the flow, if success -> create an invoice_sent document, if not -> create an invoice_sending_failed document